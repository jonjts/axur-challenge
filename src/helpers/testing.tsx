import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
