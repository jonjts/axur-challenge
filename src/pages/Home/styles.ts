import styled from 'styled-components';

export const Container = styled.div`
`;

export const CardContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;

  label {
    cursor: pointer;
  }
`;