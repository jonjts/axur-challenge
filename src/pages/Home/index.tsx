import React from 'react';
import { useHistory } from 'react-router-dom';

import Card from '../../components/Card';
import Inspections from '../Inspections';

import { Container, CardContent } from './styles';

const Home: React.FC = () => {
    const { push } = useHistory();

    return (
        <>
            <h1>Home</h1>
            <Container className='grid'>
                <div >
                    <Card
                        bounce
                        pointer
                        onClick={() => push('inspections/new')}
                    >
                        <CardContent>
                            <h3>Nova Inspeção</h3>
                            <p>Adicione uma nova keyword para iniciar o processo de web crawling.</p>
                        </CardContent>
                    </Card>
                </div>
                <div>
                    <Card
                        bounce
                        pointer
                        onClick={() => push('inspections/show')}
                    >
                        <CardContent>
                            <h3>Consultar Inspeção</h3>
                            <p>Informe o id de uma inspeção já cadastrada para verificar seus resultados.</p>
                        </CardContent>
                    </Card>
                </div>
                <Inspections />
            </Container>
        </>
    );
};

export default Home;