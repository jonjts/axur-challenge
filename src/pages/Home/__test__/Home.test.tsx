import { matchSnapshotWithProviders, renderWithProviders } from '../../../helpers/testing';

import Home from '../index';

describe('Home', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<Home />);
  });

  test('renders correctly', () => {
    renderWithProviders(<Home />);
  });
});
