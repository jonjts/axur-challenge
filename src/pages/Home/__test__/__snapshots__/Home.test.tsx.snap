// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home match with snapshot 1`] = `
Array [
  <h1>
    Home
  </h1>,
  <div
    className="sc-fubCfw cqjzZl grid"
  >
    <div>
      <div
        className="sc-hKgILt hjoULL bounce"
        onClick={[Function]}
      >
        <div
          className="sc-pFZIQ fPXgLf"
        >
          <h3>
            Nova Inspeção
          </h3>
          <p>
            Adicione uma nova keyword para iniciar o processo de web crawling.
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        className="sc-hKgILt hjoULL bounce"
        onClick={[Function]}
      >
        <div
          className="sc-pFZIQ fPXgLf"
        >
          <h3>
            Consultar Inspeção
          </h3>
          <p>
            Informe o id de uma inspeção já cadastrada para verificar seus resultados.
          </p>
        </div>
      </div>
    </div>
    <div
      className="sc-jSgupP eplsmV"
    >
      <h2>
        Inspeções Recentes
      </h2>
      <div
        className="sc-gKsewC gXHAJb grid"
      />
    </div>
  </div>,
]
`;
